mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -33,11 +34,29 @@ type TgcMetadataPayload struct {
 	PrimaryResource  string                      `json:"primary_resource"`
 }
 
-// Hardcode the Terraform resource name -> API service name mapping temporarily.
-// TODO: [tgc] read the mapping from the resource metadata files.
-var ApiServiceNames = map[string]string{
-	"google_compute_instance": "compute.googleapis.com",
-	"google_project":          "cloudresourcemanager.googleapis.com",
+// PROJECT_NUMBER instead of PROJECT_ID is in the CAI asset names for the resources in those services
+// https://cloud.google.com/asset-inventory/docs/asset-names
+var serviceWithProjectNumber = map[string]struct{}{
```

Comment:
Use the map to store the services that have PROJECT_NUMBER in the cai asset names.

Will handle other cases by modifying the resource metadata yaml file.